### PR TITLE
DT-1162: ♻️ Extract add role into router

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: backend/api/prisonApi.js [hunk 1]
```diff
@@ -31,9 +31,9 @@ const prisonApiFactory = (client) => {
       context,
       `/api/users/${username}/access-roles/caseload/${config.app.applicationCaseload}?includeAdmin=${hasAdminRole}`
     )
-  const removeRole = (context, agencyId, username, roleCode) =>
```
Comment (petergphillips) [inline backend/api/prisonApi.js:34]: not used in call so removed

File: backend/controllers/removeRole.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 const removeRoleFactory = (prisonApi) => {
   const removeRole = async (req, res) => {
-    const { agencyId, username, roleCode } = req.query
-    await prisonApi.removeRole(res.locals, agencyId, username, roleCode)
+    const { username, roleCode } = req.query
```
Comment (petergphillips) [inline backend/controllers/removeRole.js:3]: not used by call so removed

File: backend/routes/manageAuthUserRouter.js [hunk 3]
```diff
@@ -1,10 +1,21 @@
 const express = require('express')
-const { addRoleFactory } = require('../controllers/addRole')
+const { selectRolesFactory } = require('../controllers/addRole')
 
 const router = express.Router({ mergeParams: true })
 
-const controller = ({ oauthApi, prisonApi, logError }) => {
-  const { index, post } = addRoleFactory(oauthApi, prisonApi, logError)
+const controller = ({ oauthApi, logError }) => {
+  const getUserAndRoles = (context, username) =>
```
Comment (petergphillips) [inline backend/routes/manageAuthUserRouter.js:7]: plan is to add a `managePrisonUserRouter` too that then shares the same factory

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**